source/code/go/src/plugins/telemetry.go

```diff
@@ -0,0 +1,155 @@
+	metricNameAvgFlushRate       = "ContainerLogAvgRecordsFlushedPerSec"
+	defaultTelemetryPushInterval = 300
+
+	// EventNameContainerLogInit name of the event
+	EventNameContainerLogInit = "ContainerLogPluginInitialized"
+)
+
+// Initialize initializes the telemetry artifacts
+func initialize(telemetryPushIntervalProperty string, agentVersion string) (int, error) {
+
+	telemetryPushInterval, err := strconv.Atoi(telemetryPushIntervalProperty)
+	if err != nil {
+		Log("Error Converting telemetryPushIntervalProperty %s. Using Default Interval... %d \n", telemetryPushIntervalProperty, defaultTelemetryPushInterval)
+		telemetryPushInterval = defaultTelemetryPushInterval
+	}
+
+	ContainerLogTelemetryTicker = time.NewTicker(time.Second * time.Duration(telemetryPushInterval))
+
+	encodedIkey := os.Getenv(envAppInsightsAuth)
+	if encodedIkey == "" {
+		Log("Environment Variable Missing \n")
+		return -1, errors.New("Missing Environment Variable")
+	}
+
+	decIkey, err := base64.StdEncoding.DecodeString(encodedIkey)
+	if err != nil {
+		Log("Decoding Error %s", err.Error())
+		return -1, err
+	}
+
+	TelemetryClient = appinsights.NewTelemetryClient(string(decIkey))
+
+	CommonProperties = make(map[string]string)
+	CommonProperties["Computer"] = Computer
+	CommonProperties["WorkspaceID"] = WorkspaceID
+	CommonProperties["ControllerType"] = controllerTypeDaemonSet
+	CommonProperties["AgentVersion"] = agentVersion
+
+	aksResourceID := os.Getenv(envAKSResourceID)
+	// if the aks resource id is not defined, it is most likely an ACS Cluster
+	if aksResourceID == "" {
+		CommonProperties["ACSResourceName"] = os.Getenv(envACSResourceName)
+		CommonProperties["ClusterType"] = clusterTypeACS
+
+		CommonProperties["SubscriptionID"] = ""
+		CommonProperties["ResourceGroupName"] = ""
+		CommonProperties["ClusterName"] = ""
+		CommonProperties["Region"] = ""
+
+	} else {
+		CommonProperties["ACSResourceName"] = ""
+		splitStrings := strings.Split(aksResourceID, "/")
+		CommonProperties["SubscriptionID"] = splitStrings[2]
+		CommonProperties["ResourceGroupName"] = splitStrings[4]
+		CommonProperties["ClusterName"] = splitStrings[8]
+		CommonProperties["ClusterType"] = clusterTypeAKS
+
+		region := os.Getenv("AKS_REGION")
+		if region != "" {
+			CommonProperties["Region"] = region
+		}
+	}
+
+	TelemetryClient.Context().CommonProperties = CommonProperties
+	return 0, nil
+}
+
+// SendContainerLogFlushRateMetric is a go-routine that flushes the data periodically (every 5 mins to App Insights)
+func SendContainerLogFlushRateMetric(telemetryPushIntervalProperty string, agentVersion string) {
+
+	ret, err := initialize(telemetryPushIntervalProperty, agentVersion)
+	if ret != 0 || err != nil {
+		Log("Error During Telemetry Initialization :%s", err.Error())
+		runtime.Goexit()
+	}
+
+	SendEvent(EventNameContainerLogInit, make(map[string]string))
+
+	for ; true; <-ContainerLogTelemetryTicker.C {
+		flushRate := FlushedRecordsCount / FlushedRecordsTimeTaken * 1000
```

Comment:
there could be divide by zero error here

Reply:
handled by go :) 
 2018/10/12 15:32:37 oms.go:73: Flushed Records : 0.000000 Time Taken : 0.000000 flush Rate : NaN